api/login.go

```diff
@@ -29,13 +32,46 @@ type LoginResult struct {
 	TokenType   string      `json:"token_type"`
 }
 
+// LoginLog record user login info, no matter success or not
+type LoginLog struct {
+	ClientInfo
+	LoginID   string
+	LoginTime types.NanoTS
+	LoginIP   string
+	IsSuccess bool
+}
+
+func (l LoginLog) Stringer() string {
```

Comment:
https://pkg.go.dev/fmt#hdr-Printing

"5. If an operand implements method String() string, that method will be invoked to convert the object to a string, which will then be formatted as required by the verb"

What about that we just directly implement String() and directly do logrus.Infof("%v", loginLog) ?

Reply:
exactly, it's my fault, I wanted to implement Stringer Interface, and String() is method name, I was misunderstanding